---
title: "2022-10-17 Alex MS4A qPCR"
format: 
   html:
     df-print: paged
---

## Background

$\Delta CT = CT_{target.gene} - CT_{reference.gene}$

$\Delta \Delta CT = \Delta CT_{treated.sample} - \Delta CT_{control.sample} =$

$(CT_{target.gene.in.treated.sample} - CT_{reference.gene.in.treated.sample}) - (CT_{target.gene.in.control.sample} - CT_{reference.gene.in.control.sample}) =$

$CT_{target.gene.in.treated.sample} - CT_{reference.gene.in.treated.sample} - CT_{target.gene.in.control.sample} + CT_{reference.gene.in.control.sample}$

$2^{-\Delta \Delta CT}$ = fold change (FC) of target gene expression in a treated sample relative to a control sample, normalized to a reference gene

Null hypothesis:

$\Delta \Delta CT = 0$

$\text{FC} = 2^{0} = 1$

or

$\Delta CT_{treated.sample} = \Delta CT_{control.sample}$

or

$CT_{target.gene.in.treated.sample} - CT_{reference.gene.in.treated.sample} = CT_{target.gene.in.control.sample} - CT_{reference.gene.in.control.sample}$

The $2^{-\Delta \Delta CT}$ method (aka [Livak method](https://doi.org/10.1006/meth.2001.1262)) assumes a uniform PCR amplification efficiency of 100% across all samples

<https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-7-85>

## Load packages

```{r}
#| output: false
library(tidyverse)
library(magrittr)
library(readxl)
library(janitor)
library(broom.mixed)
library(lmerTest)
library(DHARMa)
library(emmeans)
library(ggeffects)
library(marginaleffects)
theme_set(theme_minimal())
```

## Load and tidy data

```{r}
df = read_csv("iMGL_iso_samples_combined_qPCR_clean.csv", show_col_types = FALSE) %>% clean_names()

df
```

```{r}
df %<>%
  rename(gene = target_name,
         ct = avg_ct,
         deltact = d_ct) %>%
  mutate(gene = fct_relevel(gene, "GAPDH"),
         genotype = fct_relevel(genotype, "CC"),
         treatment = fct_relevel(treatment, "Basal"),
         experiment = factor(experiment),
         line = factor(line),
         log2.gxp = max(ct) - ct) %>%
  select(ct, deltact, line, experiment, gene, genotype, treatment, log2.gxp)

df
```

## EDA

```{r}
df %>%
  ggplot(aes(x = gene, y = log2.gxp, color = gene, shape = genotype)) +
  geom_point(stat = "identity") +
  theme(axis.text.x = element_blank()) + # theme(axis.text.x = element_text(angle = 45, size = rel(0.5))) +
  ylab("log2(gxp)") +
  facet_wrap(~ line + experiment + treatment)
```

## MS4A4A

```{r}
df.ms4a4a = df %>% filter(gene == "MS4A4A")

fit.ms4a4a = lmer(deltact ~ 1 + genotype*treatment + (1 | line/experiment), data = df.ms4a4a)

summary(fit.ms4a4a)
```

```{r}
#| warning: FALSE
plot(simulateResiduals(fit.ms4a4a))
```

```{r}
(em.ms4a4a = emmeans(fit.ms4a4a, revpairwise ~ genotype | treatment))
```

```{r}
confint.ms4a4a = as_tibble(confint(em.ms4a4a)$contrasts) %>%
      mutate(across(c(-contrast, - treatment, -df), ~ 2^-(.x)))

confint.ms4a4a
```

```{r}
confint.ms4a4a %<>%
  left_join(as_tibble(em.ms4a4a$contrasts) %>%
              select(treatment, p.value),
            by = "treatment")

confint.ms4a4a
```

```{r}
df %>%
  filter(gene %in% c("MS4A4A", "GAPDH")) %>%
  lmer(ct ~ gene * genotype * treatment + (1 | line/experiment), data = .) %>%
  emmeans(.,  ~ gene * genotype * treatment) %>%
  contrast(., interaction = c("trt.vs.ctrl"), by = "treatment") %>%
  left_join(x = tidy(.) %>% select(treatment, p.value), y = confint(.) %>% tidy, by = "treatment") %>%
  mutate(across(c(estimate, std.error, conf.low, conf.high), ~ 2^-(.x))) %>%
  rename(conf.l = conf.high, conf.h = conf.low) %>%
  relocate(conf.l, conf.h, p.value, .after = last_col())
```

```{r}
confint.ms4a4a %>%
  ggplot(aes(x = treatment, y = estimate)) +
  geom_point(stat = "identity") +
  geom_errorbar(aes(ymin = upper.CL, ymax = lower.CL), width = .1) +
  geom_text(aes(y = lower.CL, label = ifelse(p.value < 0.001, "***", ifelse(p.value < 0.01, "**", ifelse(p.value < 0.05, "*", "")))), vjust = -0.5) +
  ylim(0, 8) +
  ylab("fold change(gene expression) [TT - CC]") +
  ggtitle("MS4A4A")
```

## MS4A6A

```{r}
df.ms4a6a = df %>% filter(gene == "MS4A6A")

fit.ms4a6a = lmer(deltact ~ 1 + genotype*treatment + (1 | line/experiment), data = df.ms4a6a)

summary(fit.ms4a6a)
```

```{r}
#| warning: FALSE
plot(simulateResiduals(fit.ms4a6a))
```

```{r}
(em.ms4a6a = emmeans(fit.ms4a6a, revpairwise ~ genotype | treatment))
```

```{r}
confint.ms4a6a = as_tibble(confint(em.ms4a6a)$contrasts) %>%
      mutate(across(c(-contrast, - treatment, -df), ~ 2^-(.x)))

confint.ms4a6a
```

```{r}
confint.ms4a6a %<>%
  left_join(as_tibble(em.ms4a6a$contrasts) %>%
              select(treatment, p.value),
            by = "treatment")

confint.ms4a6a
```

```{r}
df %>%
  filter(gene %in% c("MS4A6A", "GAPDH")) %>%
  lmer(ct ~ gene * genotype * treatment + (1 | line/experiment), data = .) %>%
  emmeans(.,  ~ gene * genotype * treatment) %>%
  contrast(., interaction = c("trt.vs.ctrl"), by = "treatment") %>%
  left_join(x = tidy(.) %>% select(treatment, p.value), y = confint(.) %>% tidy, by = "treatment") %>%
  mutate(across(c(estimate, std.error, conf.low, conf.high), ~ 2^-(.x))) %>%
  rename(conf.l = conf.high, conf.h = conf.low) %>%
  relocate(conf.l, conf.h, p.value, .after = last_col())
```

```{r}
confint.ms4a6a %>%
  ggplot(aes(x = treatment, y = estimate)) +
  geom_point(stat = "identity") +
  geom_errorbar(aes(ymin = upper.CL, ymax = lower.CL), width = .1) +
  geom_text(aes(y = lower.CL, label = ifelse(p.value < 0.001, "***", ifelse(p.value < 0.01, "**", ifelse(p.value < 0.05, "*", "")))), vjust = -0.5) +
  ylim(0, 8) +
  ylab("fold change(gene expression) [TT - CC]") +
  ggtitle("MS4A6A")
```

## MS4A4A and MS4A6A at baseline

```{r}
rbind(confint.ms4a4a %>% filter(treatment == "Basal") %>% mutate(gene = "MS4A4A"),
      confint.ms4a6a %>% filter(treatment == "Basal") %>% mutate(gene = "MS4A6A")) %>%
  ggplot(aes(x = gene, y = estimate)) +
  geom_point(stat = "identity") +
  geom_errorbar(aes(ymin = upper.CL, ymax = lower.CL), width = .1) +
  geom_text(aes(y = lower.CL, label = ifelse(p.value < 0.001, "***", ifelse(p.value < 0.01, "**", ifelse(p.value < 0.05, "*", "")))), vjust = -0.5) +
  ylim(0, 8) +
  ylab("fold change(gene expression)") +
  ggtitle("rs636317 genotype effect (TT - CC)")
  
```

## ggeffects

```{r}
df2 = df %>% filter(gene %in% c("MS4A4A", "GAPDH"))

mod = lmer(ct ~ gene * genotype * treatment + (1 | line/experiment), data = df2)

ggpredict(mod, terms = c("gene", "genotype", "treatment")) %>% plot

comparisons(mod, variables = list(gene = "reference"), by = c("genotype", "treatment"), hypothesis = "b1 = b2")
```
